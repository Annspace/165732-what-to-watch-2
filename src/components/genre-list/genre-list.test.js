import React from 'react';
import renderer from 'react-test-renderer';
import GenreList from './genre-list.jsx';
import mockData from '../../mocks/films.js';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore({
  genre: `All genres`,
  filteredMovies: mockData.movies,
});

it(`renders correctly`, () => {
  const clickGenreHandler = jest.fn();
  const setGenreHandler = jest.fn();
  const GenreListComponent = renderer
    .create(
        <Provider store={store}>
          <GenreList setGenre={setGenreHandler} genreToShow={`drama`} onclickGenre={clickGenreHandler} movies={mockData.movies}/>
        </Provider>);
  expect(GenreListComponent).toMatchSnapshot();
});
