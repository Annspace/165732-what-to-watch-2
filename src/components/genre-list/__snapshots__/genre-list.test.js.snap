// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item
           catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item
           "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
    >
      fantasy
    </a>
  </li>
  <li
    className="catalog__genres-item
           "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
    >
      drama
    </a>
  </li>
  <li
    className="catalog__genres-item
           "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
    >
      comedy
    </a>
  </li>
</ul>
`;
